BDArmory/Control/MissileFire.cs

```diff
@@ -9341,6 +9335,13 @@ void ToggleTurret()
         #endregion Turret
 
         #region Aimer
+        bool ShowBoreRing(bool visible)
+        {
+            if (!vessel.isActiveVessel) return false; // We're not in control.
+            if (!showBombAimer) return false; //tachnically, this should only be appearing when aiming bombs. Would need to edit this if a dynamic ring is ever implemented for missile boresights
```

Comment:
This would prevent hiding the `boreRing` when `showBombAimer` is `false` and tell the WM that it isn't the one controlling the `boreRing` when it actually is.
The logic for `boreRing`'s dependency on `showBombAimer` is already being handled in `OnUpdate` with the `missileAimerUI` stuff. I'm not sure how much of the rest of that should also only be visible for the active vessel. Possibly the entire `//BOMB/MISSILE HUD` section should be enclosed in an `if (vessel.isActiveVessel)` check?

Reply:
OnGUI already has everything enclosed in an active vessel check - ``if (HighLogic.LoadedSceneIsFlight && vessel == FlightGlobals.ActiveVessel && BDArmorySetup.GAME_UI_ENABLED && !MapView.MapIsEnabled)``

Reply:
Sure, but the `if (!showBombAimer) return false;` line here would prevent the `boreRing` from getting set inactive even when called from the active vessel.

I'm not sure what the `missileAimerUI` and all the other stuff in the `//BOMB/MISSILE HUD` is doing, but if it's only for the active vessel, then it's wasted computation if it's being doing for the other vessels.

Reply:
Oh, you mean the stuff that all got moved out of OnGUI to OnUpdate - yes, that should also be inside a ``if (activeVessel)`` check since it's currently running for everything. Sorry, was mistaking that for the HUD stuff still in OnGUI